packages/core/src/checkbox/checkbox.scss

```diff
@@ -52,25 +47,99 @@
   box-shadow: var(--_box-shadow);
   display: inline-flex;
   flex-shrink: 0;
-  margin-block-start: calc((var(--_line-height) - var(--_size) - var(--_border-size) * 2) / 2);
-  transition: 0.2s ease-in-out;
-  transition-property: background, border, box-shadow, color, filter;
+
+  padding: calc((var(--_size) - var(--_icon-size)) / 2 - var(--_border-size));
+  transition: var(--_transition);
+  transition-property: background, border-color, box-shadow, color, filter;
+}
+
+.label {
+  margin-block-start: calc((var(--_size) - var(--_line-height)) / 2);
 }
 
+sl-icon,
 svg {
   align-self: center;
   box-sizing: border-box;
   fill: none;
-  height: var(--_size);
+  height: var(--_icon-size);
   transition: var(--_transition);
   transition-property: background, border, box-shadow, fill;
-  width: var(--_size);
+  width: var(--_icon-size);
 }
 
 path {
   stroke: var(--_stroke);
   stroke-dasharray: var(--_path-length);
   stroke-dashoffset: var(--_stroke-dashoffset);
+  stroke-linecap: round;
```

Comment:
Shouldn't this be a css variable? Bingel = playful = round; Kampus = business = sharp?

Reply:
I've made it a variable, but there isn't a token for it. We're also not 100% sure how we're going to make the icon; svg in place or use sl-icon